python/graphstorm/gconstruct/remap_result.py

```diff
@@ -0,0 +1,477 @@
+
+        Return
+        ------
+        (int, int): start idx and end idx
+    """
+    if world_size == 0:
+        return 0, num_files
+
+    files_per_inst = num_files // world_size
+    rest = num_files % world_size
+    # evenly spread the fiels
+    # For example, with 10 files and world_size == 4
+    # number files for each rank will be:
+    # 2 2 3 3
+    if rank + rest < world_size:
+        start = files_per_inst * rank
+        end = start + files_per_inst
+    else:
+        start = files_per_inst * rank + rank + rest - world_size
+        end = start + files_per_inst + 1
+
+    return start, end
+
+def remap_edge_pred(pred_etypes, pred_dir,
+                    output_dir, out_chunk_size,
+                    num_proc, rank, world_size, with_shared_fs,
+                    preserve_input=False):
+    """ Remap edge prediction result.
+
+        The function wil iterate all the edge types that
+        have prediction results and spin num_proc workers
+        to do the rampping jos.
+
+        The directory storing prediction results looks like:
+        # Predicionts:
+        #    predict-00000.pt
+        #    predict-00001.pt
+        #    ...
+        #    src_nids-00000.pt
+        #    src_nids-00001.pt
+        #    ...
+        #    dst_nids-00000.pt
+        #    dst_nids-00001.pt
+        #    ...
+
+        The output emb files will be
+        #    predict-00000_00000.parquet
+        #    predict-00000_00001.parquet
+        #    ...
+
+        Parameters
+        ----------
+        pred_etypes: list of tuples
+            List of edge types that have prediction results to be remapped。
+        pred_dir: str
+            The directory storing the prediction results.
+        output_dir: str
+            The directory storing the remapped prediction results.
+        out_chunk_size: int
+            Max number of raws per output file.
+        num_proc: int
+            Number of workers used in processing.
+        rank: int
+            The global rank of current processes.
+        world_size: int
+            The total number of processes in the cluster.
+        with_shared_fs: bool
+            Whether shared file system is avaliable
+        preserve_input: bool
+            Whether the input data should be removed.
+    """
+    start = time.time()
+    task_list = []
+    for etype in pred_etypes:
+        input_pred_dir = os.path.join(pred_dir, "_".join(etype))
+        out_pred_dir = os.path.join(output_dir, "_".join(etype))
+        etype_pred_files = os.listdir(input_pred_dir)
+        src_nid_files = [fname for fname in etype_pred_files if fname.startswith("src_nids")]
+        dst_nid_files = [fname for fname in etype_pred_files if fname.startswith("dst_nids")]
+        pred_files = [fname for fname in etype_pred_files if fname.startswith("predict")]
+        src_nid_files.sort()
+        dst_nid_files.sort()
+        pred_files.sort()
+
+        num_parts = len(pred_files)
```

Comment:
should you verify if src_nid_files and dst_nid_files have the same number of files?

Reply:
Good point